### DISCUSSION TITLE
Schema file?

### DISCUSSION THREAD

**Discussion by garlandcrow** [pocketbase/pocketbase#816] [Q&A] [answered]
Am I missing something or is the only way to define collection schemas by using the web UI? There's no way commit the DB design to git? How are people handling this then? Do I make like a .sql script that sets up the SQLite db? Do I make some script that calls the API to set everything up?

**Comment by ganigeorgiev** [comment, marked as answer]
Currently, if you use the prebuilt PocketBase application, then Yes - you can define the schema from the UI, export it to JSON from "Settings > Export collections" and commit this file. You can import the JSON back from "Settings > Import collecitons".

If you use PocketBase as framework, there is automated command `migrate collections` that will generate a db migration with a snapshot of your current collection changes - https://pocketbase.io/docs/db-migrations/#collections-migration.

This certainly is not ideal and the flow will be improved in the near future (most likely with v0.9 or v0.10 release). You can also check this discussion - https://github.com/pocketbase/pocketbase/discussions/489.

**Reply by ganigeorgiev** [reply]
@garlandcrow Just to let you know that with [PocketBase v0.9.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.9.0) there is now support for automigrations, which means that any collections change will result in a `pb_migrations/*` file that could be committed and added to your repo.
You can also create custom external migrations using JavaScript (_more documentation will be available soon_).
